tribits/core/package_arch/TribitsPackageDependencies.cmake

```diff
@@ -0,0 +1,204 @@
+#     DEPENDENCIES <upstreamTpl_0> <upstreamTpl_1>:<vis_1> ... )
+#
+# The listed upstream dependencies ``<upstreamTpl_i>`` are other external
+# packages/TPLs listed before this external packages/TPL ``<tplName>`` in a
+# `<repoDir>/TPLsList.cmake`_ file.  Each upstream dependency can include a
+# visibility specification ``<vis_i>`` that can be added to the dependency
+# using a colon ``:`` with ``<upstreamTpl_1>:<vis_1>`` where ``<vis_i>`` can
+# take the allowed values:
+#
+# * ``PUBLIC``: The include directories along with the libraries will be
+#   propagated downstream to clients of ``<tplName>``.
+#
+# * ``PRIVATE``: Only the libraries will be propagated downstream to clients
+#   of ``<tplName>``.
+#
+# If ``<vis_i>`` is not specified, then the default is ``PRIVATE``.  (If a
+# package needs the include directories from some external package/TPL, then
+# it should list that external package/TPL as a direct dependency and not
+# expect to get include directories from indirect dependencies.)
+#
+macro(tribits_external_package_define_dependencies
+    tplName
+  )
+
+  # Parse arguments
+  cmake_parse_arguments(
+     PARSE "" "" # prefix, options, one_value_keywords
+     "DEPENDENCIES"  # multi_value_keywords
+     ${ARGN}
+     )
+  tribits_check_for_unparsed_arguments(PARSE)
+  tribits_assert_parse_arg_one_or_more_values(PARSE  DEPENDENCIES)
+
+  set(libAllDependencies "")
+  foreach (upstreamTplDepEntry  IN  LISTS  PARSE_DEPENDENCIES)
+    list(APPEND libAllDependencies ${upstreamTplDepEntry})
+  endforeach()
+
+  set(${tplName}_LIB_ALL_DEPENDENCIES  ${libAllDependencies}  CACHE STRING
+    "List of all potential dependencies for external package/TPL '${tplName}'")
+  mark_as_advanced(${tplName}_LIB_ALL_DEPENDENCIES)
+
+endmacro()
+
+
+# @MACRO: tribits_external_package_setup_enabled_dependencies()
+#
+# Usage::
+#
+#   tribits_external_package_setup_enabled_dependencies(<externalPkgName>)
+#
+# Macro that sets up the list of enabled external package/TPL dependencies
+#
+# Takes the list ``<externalPkgName>_LIB_ALL_DEPENDENCIES`` and sets the
+# default entries of the non-cache var
+# ``<externalPkgName>_LIB_ENABLED_DEPENDENCIES``.  However, if
+# ``${<externalPkgName>_LIB_ALL_DEPENDENCIES}`` is non-empty when this macro
+# is called, then it will not be changed.  That allows the user to override
+# the list of enabled TPL dependencies in the cache.  This also sets the
+# non-cache vars ``<externalPkgName>_ENABLE_<upstsreamPkgName>=ON`` for each
+# enabled package listed in ``<externalPkgName>_LIB_ENABLED_DEPENDENCIES`` and
+# to ``OFF`` for each ``<upstsreamPkgName>`` listed in
+# ``<externalPkgName>_LIB_ENABLED_DEPENDENCIES`` but not in
+# ``<externalPkgName>_LIB_ENABLED_DEPENDENCIES``.
+#
+macro(tribits_external_package_setup_enabled_dependencies  externalPkgName)
+
+  set(libEnabledDependencies "")
+
+  if (TPL_ENABLE_${externalPkgName})
+    foreach(upstreamPkgEntry  IN  LISTS ${externalPkgName}_LIB_ALL_DEPENDENCIES)
```

Comment:
In this case, I don't need it because of the delimiter `${`.  Therefore, I can easily see the space between `... LISTS ${external...`.

I am finding it hard to select a style that is readable and looks good in all cases.